config/v1/types_infrastructure.go

```diff
@@ -52,10 +52,22 @@ type InfrastructureStatus struct {
 	// For more info: https://github.com/etcd-io/etcd/blob/329be66e8b3f9e2e6af83c123ff89297e49ebd15/Documentation/op-guide/clustering.md#dns-discovery
 	EtcdDiscoveryDomain string `json:"etcdDiscoveryDomain"`
 
-	// apiServerURL is a valid URL with scheme(http/https), address and port.
-	// apiServerURL can be used by components like kubelet on machines, to contact the `apisever`
-	// using the infrastructure provider rather than the kubernetes networking.
-	APIServerURL string `json:"apiServerURL"`
+	// apiServerURL is a URL for the Kubernetes API.
+	//
+	// Deprecated: Use apiServerExternalURI or apiServerInternalURI instead.
+	// +optional
+	APIServerURL string `json:"apiServerURL,omitempty"`
+
+	// apiServerExternalURI is a valid URI with scheme(http/https),
+	// address and port.  apiServerExternalURI can be used by components
+	// like the web console to tell users where to find the Kubernetes API.
+	APIServerExternalURI string `json:"apiServerExternalURI"`
+
+	// apiServerInternalURI is a valid URI with scheme(http/https),
+	// address and port.  apiServerInternalURI can be used by in-cluster
+	// components like the machine-config operator to tell the kubelet
+	// where to find the Kubernetes API.
+	APIServerInternalURI string `json:"apiServerInternalURI"`
```

Comment:
I would prefer to only add this field and redefine apiServerURL to be external.  Most clients will use external.  Only a few clients use internal.

Reply:
Most internal users should use kubernetes.default.svc (which they already do), so internal is limited to a small subset of use cases.

Reply:
> I would prefer to only add this field and redefine apiServerURL to be external.

Done in b47d12f9 -> a33660b8.